packages/labs/context/src/test/std-decorators/context-provider_test.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {LitElement, html, TemplateResult} from 'lit';
+import {property} from '@lit/reactive-element/std-decorators.js';
+
+import {createContext} from '@lit-labs/context';
+import {consume, provide} from '@lit-labs/context/std-decorators.js';
+import {assert} from '@esm-bundle/chai';
+import {memorySuite} from './test_util.js';
+
+const simpleContext = createContext<number>('simple-context');
+const optionalContext = createContext<number | undefined>('optional-context');
+
+class ContextConsumerElement extends LitElement {
+  @consume({context: simpleContext, subscribe: true})
+  @property({type: Number})
+  public accessor value: number | undefined;
+
+  @consume({context: simpleContext, subscribe: true})
+  @property({})
+  public accessor value2: number | undefined = undefined;
+
+  @consume({context: optionalContext, subscribe: true})
+  @property({type: Number})
+  public accessor optionalValue: number | undefined;
```

Comment:
`@consume` isn't written to require `accessor`. Should we add a test that decorates a plain field? It should still update because the ContextConumer calls requestUpdate, yeah?